hudi-utilities/src/main/java/org/apache/hudi/utilities/schema/HiveSchemaProvider.java

```diff
@@ -52,12 +52,11 @@
   private final Schema sourceSchema;
   private Schema targetSchema;
 
-  public HiveSchemaProvider(TypedProperties props, JavaSparkContext jssc) {
-    super(props, jssc);
+  public HiveSchemaProvider(TypedProperties props, SparkSession spark) {
+    super(props, JavaSparkContext.fromSparkContext(spark.sparkContext()));
     DataSourceUtils.checkRequiredProperties(props, Collections.singletonList(Config.SOURCE_SCHEMA_TABLE_PROP));
     String sourceSchemaDatabaseName = props.getString(Config.SOURCE_SCHEMA_DATABASE_PROP, "default");
     String sourceSchemaTableName = props.getString(Config.SOURCE_SCHEMA_TABLE_PROP);
-    SparkSession spark = SparkSession.builder().config(jssc.getConf()).enableHiveSupport().getOrCreate();
```

Comment:
> does this create a new sparkSession rather than fetching an existing one and hence we had to do all the refactoring with JSSC and spark session is it ?

It created a new one, with this sparkSession created inside HiveSchemaProvider i can not obtain glue data.
I guess we need to refactor the jssc and sparksession, a big move.